Dynamic project and yearbook pages

craig-riecke opened the pull request:
- Removed Register button to make room for Projects menu item
- Added Projects page, populated by data in meta repo
- Added Help Wanted page
- Added 2017 Yearbook projects, populated dynamically

craig-riecke committed:
Dynamic project and yearbook pages

seanmarcia commented on _includes/_global_js.html:
```diff
@@ -0,0 +1,55 @@
+  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+
+  ga('create', 'UA-77346638-1', 'codefordc.org');
```
Probably don't want to put in code for dc's google analytics ;)

craig-riecke replied:
Good point!

csexton replied:
I'd just drop the google analytics all together.

craig-riecke commented:
I tried to fork this repo, but could only get it to host as https://craig-riecke.github.io/rubyforgood.html, which made all the assets and links not work.  So you'll need to set up a local host to see what it actually looks like.

csexton commented on _includes/_global_js.html:
```diff
@@ -0,0 +1,55 @@
+    });
+</script>
+
+{% if page.deeplink == true %}
```
I think we can remove this entire section, no pages currently set `deeplink: true`.

csexton commented on _includes/_global_js.html:
```diff
@@ -0,0 +1,55 @@
+
+          }
+        });
+    });
```
Local testing it doesn't look like this works/is used. I'd prefer to just delete this snippet.

csexton commented on _includes/_global_js.html:
```diff
@@ -0,0 +1,55 @@
+<!-- This should be inserted in the page *before* any page-specific JS -->
+<script src="/assets/js/jquery.js"></script>
+<script src="/assets/js/bootstrap.min.js"></script>
+{% for js_name in page.extra_js %}<script src="/assets/js/{{ js_name }}"></script>{% endfor %}
```
Any reason we shouldn't just include the `extra_js` files from the project page in the `base.html` layout?

csexton commented on _includes/_global_js.html:
```diff
@@ -0,0 +1,55 @@
+<!-- This should be inserted in the page *before* any page-specific JS -->
+<script src="/assets/js/jquery.js"></script>
+<script src="/assets/js/bootstrap.min.js"></script>
```
Duplicate includes, both bootstrap and jquery are already included in the `base.html` layout.

csexton commented on _includes/projects-yearbook.html:
```diff
@@ -0,0 +1,75 @@
+
+<div class="clearfix"></div>
+
+{% include _global_js.html %}
```
Looks like there are duplicate includes in this file.

csexton commented on _layouts/base.html:
```diff
@@ -24,6 +24,7 @@
   <link rel="stylesheet" href="assets/css/jquery.fancybox.css" type="text/css" media="all" />
   <link rel="stylesheet" href="assets/css/style-green.css" id="mainstyle" type="text/css" media="all" />
   <link rel="stylesheet" href="assets/css/custom.css" id="custom" type="text/css" media="all" />
+  {% for css_name in page.extra_css %}<link rel="stylesheet" href="/assets/css/{{ css_name }}">{% endfor %}
```
Just like the JS: No need to special case the CSS includes, we should just load them all in `base.html`

csexton commented on helpwanted.html:
```diff
@@ -0,0 +1,69 @@
+    }
+  });
+};
+</script>
```
I love this page! Nice work :heart:

csexton reviewed (changes requested):
Ran things locally and it all looked great. This is really awesome I just have a few nit-picky things I'd like to get cleaned up. 

The projects, yearbook and help wanted pages are absolutely wonderful.

craig-riecke committed:
Fixed review issues and added slack channel

craig-riecke commented:
All righty, I took all of your suggestions.  It turns out the _global.js stuff was funky due to page loading timing issues.  Once I fixed those, _global.js wasn't needed anymore so I removed it.   Lemme know if anything is still amiss.

csexton reviewed (approved):
Thanks for the updates @craig-riecke, much simpler. I ran everything locally and it all looked great -- even the new slack section on the project page.

:shipit:
